api/nodes/views.py

```diff
@@ -296,37 +295,23 @@ def bulk_destroy_skip_uneditable(self, resource_object_list, user, object_type):
 
     # Overrides BulkDestroyModelMixin
     def perform_bulk_destroy(self, resource_object_list):
-
-        auth = get_user_auth(self.request)
-        date = timezone.now()
-        id_list = [x.id for x in resource_object_list]
-
-        if NodeRelation.objects.filter(
-            parent__in=resource_object_list,
-            child__is_deleted=False,
-        ).exclude(Q(child__in=resource_object_list) | Q(is_node_link=True)).exists():
-            raise ValidationError('Any child components must be deleted prior to deleting this project.')
-
-        remove_addons(auth, resource_object_list)
+        if enforce_no_children(self.request):
+            if NodeRelation.objects.filter(
+                parent__in=resource_object_list,
+                child__is_deleted=False,
+            ).exclude(Q(child__in=resource_object_list) | Q(is_node_link=True)).exists():
+                raise ValidationError('Any child components must be deleted prior to deleting this project.')
 
         for node in resource_object_list:
-            node.add_remove_node_log(auth=auth, date=date)
-
-        nodes = AbstractNode.objects.filter(id__in=id_list)
-        nodes.update(is_deleted=True, deleted_date=date)
-        if nodes.filter(is_public=True).exists():
-            AbstractNode.bulk_update_search(resource_object_list)
-        for node in nodes:
-            project_signals.node_deleted.send(node)
+            self.perform_destroy(node)
```

Comment:
`BulkDestroyModelMixin.perform_bulk_destroy` calls `perform_destroy`, but our overrides weren't doing that, just taking care of deleting the node within the `perform_bulk_destroy` method.  Now, I am calling perform_destroy for each node.